crates/ibc/src/core/ics04_channel/commitment.rs

```diff
@@ -109,8 +109,32 @@ pub(crate) fn compute_ack_commitment(ack: &Acknowledgement) -> AcknowledgementCo
 ///
 /// Note that computing commitments with anything other than SHA256 will
 /// break the Merkle proofs of the IBC provable store.
-fn hash(data: impl AsRef<[u8]>) -> Vec<u8> {
+fn hash(data: &[u8]) -> Vec<u8> {
```

Comment:
Given the above suggestions, this `hash` function should accordingly return: `[u8; 32]`